node/src/components/consensus/protocols/highway.rs

```diff
@@ -153,7 +153,8 @@ impl<I: NodeIdT, C: Context> HighwayProtocol<I, C> {
                                 ));
                             } else {
                                 // It's not a block: Add it to the state.
-                                results.extend(self.add_valid_vertex(vv, rng));
+                                let now = Timestamp::now();
```

Comment:
As a next step (not necessarily this PR), this should be passed in here (and in `resolve_validity`), too, so that the `now()` call is made in `era_supervisor.rs` instead.

Reply:
Shouldn't the timestamp for the vertex be chosen earlier? If it's "our vertex" it should come from the `BlockContext` and for an incoming vertex it should be part of it.

Reply:
Why? It doesn't make a difference whether we call `now()` here, or pass in `now()` as an argument.

(The timestamp we want to pass into `Highway` here is not the vertex', but _our_ current timestamp. E.g. if we received a proposal and are now making a confirmation, the confirmation should contain our own current time as its timestamp, not the proposal's.)